### DISCUSSION TITLE
Field type otp

### DISCUSSION THREAD

**Discussion by icodus** [buttercup/buttercup-desktop#1364] [General]
What should I put into field-type.otp in bettercup-desktop? Tried all combinations of otpauth but nothing worked. Would be grateful for a help.

**Comment by zicklag** [comment]
You can either use a Desktop QR code scanner to extract the OTP URL from the QR code used for most 2FA configuration menus, or you can create a URL yourself by taking the secret that they give you and substituting it into the URL like this:

    otpauth://totp/generator?period=30&digits=6&secret=SecretFromOTPSetupPage

More info in this issue: https://github.com/buttercup/buttercup-desktop/issues/1042

Looking around, annoyingly there's no standard specification for the otpauth URLs: https://shkspr.mobi/blog/2022/05/why-is-there-no-formal-specification-for-otpauth-urls/

You shouldn't really need to mess with the URL most of the time I don't think, but here is one of the closer things we have to a spec, and the list of options if you want to check it out:

https://github.com/google/google-authenticator/wiki/Key-Uri-Format

The biggest reason you might want change the URL is to replace `generator` in that example URL above with a friendly name associated to the app you are trying to login with, so that it will show up as like `GitHub` instead of `generator` in the OTP password list.
